src/lib/validateFrontmatter.ts

```diff
@@ -0,0 +1,78 @@
+import type DocsMigrateCommand from '../commands/docs/migrate.js';
+import type DocsUploadCommand from '../commands/docs/upload.js';
+import type { GuidesRequestRepresentation } from '../types.js';
+import type { PageMetadata } from './readPage.js';
+
+import ora from 'ora';
+
+import { fix, writeFixes } from './frontmatter.js';
+import isCI from './isCI.js';
+import { oraOptions } from './logger.js';
+import promptTerminal from './promptWrapper.js';
+import { fetchMappings, fetchSchema } from './readmeAPIFetch.js';
+
+export async function validateFrontmatter(
+  this: DocsMigrateCommand | DocsUploadCommand,
+  pages: PageMetadata[],
+  promptQuestion: string,
+  outputDir?: string,
+): Promise<PageMetadata<GuidesRequestRepresentation>[]> {
+  const { path: pathInput } = this.args;
+  let pagesToReturn = pages;
+
+  const validationSpinner = ora({ ...oraOptions() }).start('🔬 Validating frontmatter data...');
+
+  const schema = await fetchSchema.call(this);
+  const mappings = await fetchMappings.call(this);
+
+  // validate the files, prompt user to fix if necessary
+  const validationResults = pages.map(file => {
+    this.debug(`validating frontmatter for ${file.filePath}`);
+    return fix.call(this, file.data, schema, mappings);
+  });
+
+  const filesWithIssues = validationResults.filter(result => result.hasIssues);
+  this.debug(`found ${filesWithIssues.length} files with issues: ${JSON.stringify(filesWithIssues)}`);
+  const filesWithFixableIssues = filesWithIssues.filter(result => result.fixableErrorCount);
+  const filesWithUnfixableIssues = filesWithIssues.filter(result => result.unfixableErrors.length);
+
+  if (filesWithIssues.length) {
```

Comment:
Since the bulk of this method is in this block, how do you feel about flipping it?

```suggestion
  if (!filesWithIssues.length) {
    validationSpinner.succeed(`${validationSpinner.text} no issues found!`);
    return pagesToReturn
  }
```

Reply:
done! https://github.com/readmeio/rdme/pull/1233/commits/bb79f25843457e999d701f139670f901c83b93bb